_rules/table-header-cell-has-assigned-cells-d0f69e.md

```diff
@@ -23,36 +23,42 @@ acknowledgments:
   authors:
     - Audrey Maniez
     - Jey Nandakumar
+    - Aron Janecki
 ---
 
 ## Applicability
 
-This rule applies to any HTML element with the [semantic role][] of [rowheader][] or [columnheader][] for which all of the following is true:
+This rule applies to any [HTML element][] with the [semantic role][] of [rowheader][] or [columnheader][] that is [included in the accessibility tree][] and for which all of the following are true:
 
-- the element is [visible][]; and
-- the element is [included in the accessibility tree][]; and
-- the element has at least one ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][]; and
-- the element's closest ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][] is [included in the accessibility tree][].
+- **in a row** the element is a child of an [accessible object][] with the [semantic role][] of [row][]; and
+- **table** the element is an [owned element][] of an [accessible object][] with the [semantic role][] of either [table][] or [grid][] that is [visible][]; and
```

Comment:
The definition of "owned by" does not fit very well in here. A `row` can be owned by a `table` or it can be owned by `rowgroup` that is owned by a `table`.  I wanted to catch both.

Reply:
What about inclusive descendants?